### ISSUE TITLE
Cannot transfer Batch Normalization to a tensorflow BN operator

### ISSUE THREAD

**Issue by LexXia** [onnx/onnx-tensorflow#356]
Hi all. I'm transferring a typical resnet50 model (from pytorch torchvision.model) to a tensorflow-gpu model. The model's version is IR:3 and OPSET:9. After transformation, I find that the BN operator is replaced by mutiple adding and multiplication operators with several splitting operators. I have tested the speed of the obtained tensorflow model, whose inference latency is about 1000X higher than an original tensorflow model using tensorflow BN operator. Is that a common result?
Some of my configurations
Python version:
3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34)
[GCC 7.3.0]
ONNX version:
1.3.0
ONNX-TF version:
1.2.1
Tensorflow version:
1.9.0

**Comment by fumihwh** [comment]
@LexXia 
```
import tensorflow as tf

from onnx_tf.handlers.backend_handler import BackendHandler
from onnx_tf.handlers.handler import onnx_op
from onnx_tf.handlers.handler import tf_func


@onnx_op("BatchNormalization")
@tf_func(tf.nn.fused_batch_norm)
class BatchNormalization(BackendHandler):

  @classmethod
  def get_attrs_processor_param(cls):
    return {
        "default": {
            "epsilon": 1e-5
        },
    }

  @classmethod
  def _common(cls, node, **kwargs):
    tensor_dict = kwargs["tensor_dict"]
    x = tensor_dict[node.inputs[0]]
    x_shape = x.get_shape().as_list()
    x_rank = len(x_shape)

    params_shape_broadcast = list([1, x_shape[1]] +
                                  [1 for _ in range(2, x_rank)])

    total_num_dim = len(x.get_shape())
    scale = tf.reshape(tensor_dict[node.inputs[1]], params_shape_broadcast)
    bias = tf.reshape(tensor_dict[node.inputs[2]], params_shape_broadcast)
    running_mean = tf.reshape(tensor_dict[node.inputs[3]],
                              params_shape_broadcast)
    running_variance = tf.reshape(tensor_dict[node.inputs[4]],
                                  params_shape_broadcast)

    # from version 7, force to use test mode
    if cls.SINCE_VERSION >= 7 or node.attrs.get("is_test", 0):
      inputs = [x, scale, bias, running_mean, running_variance]
      return [cls.make_tensor_from_onnx_node(node, inputs=inputs)]
    spatial = node.attrs.get("spatial", 1) == 1
    momentum = node.attrs.get("momentum", 0.9)
    axis = [0] if spatial else [0] + list(range(2, total_num_dim))
    mean, variance = tf.nn.moments(x, axis)
    running_mean = running_mean * momentum + mean * (1 - momentum)
    running_variance = running_variance * momentum + variance * (1 - momentum)
    # TODO: need to conform to the documentation here
    inputs = [x, scale, bias, running_mean, running_variance]
    return [cls.make_tensor_from_onnx_node(node, inputs=inputs)]

  @classmethod
  def version_1(cls, node, **kwargs):
    return cls._common(node, **kwargs)

  @classmethod
  def version_6(cls, node, **kwargs):
    return cls._common(node, **kwargs)

  @classmethod
  def version_7(cls, node, **kwargs):
    return cls._common(node, **kwargs)
```
Replace with `onnx_tf/handlers/backend/batch_normalization.py`
I didn't try it, plz tell me the result.

**Comment by AlexVeuthey** [comment]
@fumihwh If I understood your solution correctly, it implied replacing the `batch_normalization.py` code with your proposition? I just tried it and the resulting tensorflow model still showed the weird add/reshape/multiplication instead of normal FusedBatchNorm tensorflow operator... 

I've included a tensorboard screenshot below for illustration.
![batch_norm](https://user-images.githubusercontent.com/8789882/58329525-10b25580-7e35-11e9-8555-e26f983f4d6d.png)

**Comment by ftaralle** [comment]
Hi, same "slow" result found onnx BatchNorm -> Lots of opperations.
Moreover, i want to use the resulting TF network on GPU-android that, it seems, dosnt support Transpose opperators that are added.

Any working solution ?

**Comment by jojivk** [comment]
Hi, I am facing the same issue. Is there a way to make onnx-tensorflow generate batchnorm op. I see that onnxtokeras can generate the batchnorm,  but not onnx-tf. Any help appreciated
